Migrate java rules to pmd 7.x.x #309: common_std rule number 2: Avoid…

stokpop opened the pull request:
…CalendarDateCreation

The second common_std rule. Note this is a branch that merges into the rule number 1 branch, because it depends on `common_std` being available. So first merge rule 2 into 1 and next merge 1+2 into originating `pmd7-upgrade-7.2.0` branch.

Added a check for `Bar.getInstance().getTime()`: should not be a violation.

I noticed that match on `DateTime` class is by `ClassType/@SimpleName`, so matches both `joda` and `java.time` packages.  Can be made more specific using `typeIs`, but should we do that now?

jborgers commented:
Hmm, ik heb die andere al gemerged. common_std is er nu bij in base branch, dan moet het met deze toch goed gaan?

stokpop commented:
Added typeIs and matchesSig. Added unit test lines:

```
                DateTime nowDT3 = new DateTime(Calendar.getInstance().getTime()); // bad
                DateTime nowDT4 = new DateTime(new Date().getTime()); // good
```

The first line now has a "double" hit: see unit test with two times expected line 8. Is that an issue or can we leave that?

jborgers reviewed (approved):
Prima zo
